addOns/custompayloads/src/test/java/org/zaproxy/zap/extension/custompayloads/CustomPayloadsParamUnitTest.java

```diff
@@ -0,0 +1,140 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.custompayloads;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import org.apache.commons.configuration.HierarchicalConfiguration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.zaproxy.zap.utils.ZapXmlConfiguration;
+
+/** Unit test for {@link CustomPayloadsParam}. */
+class CustomPayloadsParamUnitTest {
+
+    private CustomPayloadsParam param;
+    private ZapXmlConfiguration configuration;
+
+    @BeforeEach
+    void setUp() {
+        param = new CustomPayloadsParam();
+        configuration = new ZapXmlConfiguration();
+    }
+
+    @Test
+    void shouldHaveConfigVersionKey() {
+        // Given / When / Then
+        assertThat(param.getConfigVersionKey(), is(equalTo("custompayloads[@version]")));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"true", "false"})
+    void shouldLoadConfirmRemoveFromConfig(boolean state) {
+        // Given
+        configuration.setProperty(CustomPayloadsParam.CONFIRM_REMOVE_PAYLOAD_KEY, state);
+        // When
+        param.load(configuration);
+        // Then
+        assertThat(param.isConfirmRemoveToken(), is(equalTo(state)));
+    }
+
+    @Test
+    void shouldDefaultConfirmRemoveTrue() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.isConfirmRemoveToken(), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldNotHaveNextPayloadIdOnUpdateFromUnversioned() {
+        // Given
+        String configKey = "custompayloads.nextPayloadId";
+        configuration = new ZapXmlConfiguration();
+        configuration.setProperty(configKey, 72);
+        param.load(configuration);
+        // When / Then
+        assertThrows(NoSuchElementException.class, () -> configuration.getInt(configKey));
+    }
+
+    @Test
+    void shouldRemoveIdsFromCustomPayloadsOnUpdate() {
+        // Given
+        String configKey = "custompayloads.categories.category(0).payloads.payload(0).payload.id";
+        createTestConfig(configuration);
+        // Before update there should be a payload with ID == 0
+        assertThat(configuration.getInt(configKey), is(equalTo(0)));
+        // When
+        param.updateConfigsImpl(-1);
+        // Then
+        assertThrows(NoSuchElementException.class, () -> configuration.getInt(configKey));
+    }
+
+    private void createTestConfig(ZapXmlConfiguration config) {
+        param.load(config);
```

Comment:
I switched this locally to a param object again because it simplified the creation of the config. I can switch it back to the manual build if that's really what you want.

Reply:
I'd say it's preferable to use what was/is expected from a previous version not what it might be currently.